src/useHotkeyState.ts

```diff
@@ -17,22 +17,72 @@ export type HotkeyShortcuts = {
  * Creates a global state singleton.
  */
 const createStateHook = () => {
+  const mousetraps = new Map<HTMLElement | undefined, MousetrapInstance>();
   let keys: HotkeyShortcuts[] = [];
 
   const addKeys = (nextKeys: HotkeyShortcuts[]) => {
     keys = [...keys, ...nextKeys];
 
     nextKeys.forEach((k) => {
-      if (!k.disabled) {
-        const mousetrap = k.ref?.current ? Mousetrap(k.ref.current) : Mousetrap;
-        mousetrap.bind(k.keys, k.callback, k.action);
+      if (k.disabled) {
+        return;
+      }
+
+      if (k.ref) {
+        if (!k.ref.current) {
+          // exit early if ref is provided but null
+          // we do not want to attach global event handlers in this case
```

Comment:
If the ref is attached asynchronously without triggering a render, the hook would not be executed again and the event wouldn't be added.